Frontend/ui-driver/src/Screens/HomeScreen/View.purs

```diff
@@ -348,25 +349,89 @@ driverMapsHeaderView push state =
                       ][ statsModel push state
                        , expandedStatsModel push state
                       ]
-                      , if not state.props.rideActionModal && (state.props.driverStatusSet == Online || state.props.driverStatusSet == Silent)  then updateLocationAndLastUpdatedView state push else dummyTextView
                   ]
                 , offlineNavigationLinks push state
-              ] <> getCarouselView (DA.any (_ == state.props.driverStatusSet) [ST.Online, ST.Silent]) false  --maybe ([]) (\item -> if DA.any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithCustomer] && DA.any (_ == state.props.driverStatusSet) [ST.Online, ST.Silent] then [] else [bannersCarousal item state push]) state.data.bannerData.bannerItem
-                <> [gotoRecenterAndSupport state push]
+              ] <> [gotoRecenterAndSupport state push]
+                <> if state.props.specialZoneProps.nearBySpecialZone then getCarouselView true false else getCarouselView (DA.any (_ == state.props.driverStatusSet) [ST.Online, ST.Silent]) false  --maybe ([]) (\item -> if DA.any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithCustomer] && DA.any (_ == state.props.driverStatusSet) [ST.Online, ST.Silent] then [] else [bannersCarousal item state push]) state.data.bannerData.bannerItem
```

Comment:
is getCarouselView required in line 364 as well?

Reply:
yes both the places

Reply:
yes